teps/0044-composing-tasks-with-tasks.md

```diff
@@ -0,0 +1,179 @@
+    overhead of using a Pipeline, they just want to make a TaskRun,
+    e.g. [@mattmoor's feedback on PipelineResources and the Pipeline beta](https://twitter.com/mattomata/status/1251378751515922432))
+    where he wants to checkout some code and [use the kaniko task](https://github.com/tektoncd/catalog/tree/master/task/kaniko/0.1)
+    without having to fiddle with volumes
+
+### Use Cases (optional)
+
+- A user wants to use catalog Tasks to checkout code, run unit tests and upload results,
+  and does not want to incur the additional overhead (and performance impact) of creating
+  volume based workspaces to share data between them in a Pipeline. e.g. specifically
+  cloning with [git-clone](https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.2),
+  running tests with [golang-test](https://github.com/tektoncd/catalog/tree/master/task/golang-test/0.1)
+  and uploading results with [gcs-upload](https://github.com/tektoncd/catalog/tree/master/task/gcs-upload/0.1).
+- An organziation does not want to use PVCs at all; for example perhaps they have decided
+  on uploading to and downloading from buckets in the cloud (e.g. GCS)
+- An organization is willing to use PVCs to some extent but needs to put limits on their use
+
+### Overlap with TEP-0046
+
+This TEP covers very similar ground as
+[TEP-0046 (colocation of tasks and workspaces)](https://github.com/tektoncd/community/pull/318), and it may be that we
+choose one solution to address both, but we have kept them separate because they are approaching the problem from
+slightly different angles.
+
+Where they are similar:
+* **Sharing data between Tasks efficiently** is the core problem at the heart of both (in this TEP, we talk about how
+  combining Tasks requires a PVC, in the other we talk about challenges around co-locating pods to share data and
+  inefficiencies of PVCs)
+
+Where they differ:
+
+* **Concurrency** (running of Tasks in parallel) is a big concern in TEP-0046 but isn't directly in the scope of this
+  problem (we could imagine a solution to this problem that only applies when running Tasks serially)
+* **The Affinity Assistant** is another focus of TEP-0046 (i.e. revisiting it before v1) but isn't directly relevant to
+  this TEP (we might solve this without changing or discussing the affinity assistant at all)
+* **Pipelines** - TEP-0046 is specifically concerned with Pipelines; there's a good chance the solution to this TEP
+  will also involve Pipelines, but there are possible solutions that could involve introducing a new type
+* **Controller configuration vs. Authoring time** - This TEP assumes that we'd want to express this kind of composition
+  at "authoring time", i.e. make it [reusable](https://github.com/tektoncd/community/blob/main/design-principles.md#reusability)
+  vs configured at runtime or at the controller level., while TEP-0046 is suggesting to configure this at the controller
+  level.
```

Comment:
Do we agree on the following ?
- Authoring time: writing `Task` or `Pipeline` (definition types)
- Runtime _time_: executing those `Task`/`Pipeline`, thus writing `PipelineRun` or `TaskRun`
- Controller level : an option on the controller to change behavior

Reply:
yep that's right! https://github.com/tektoncd/community/blob/main/design-principles.md#reusability tries to define runtime vs authoring time, let me know if you think we should reword that to make it more clear (we could add controller level too, if its not already clear)